core/frontend/src/Viewport.ts

```diff
@@ -1317,9 +1317,14 @@ export abstract class Viewport implements IDisposable {
   /** Set the display of the OpenStreetMap worldwide building layer in this viewport by attaching or detaching the reality model displaying the buildings.
    * The OSM buildings are displayed from a reality model aggregated and served from Cesium ion.<(https://cesium.com/content/cesium-osm-buildings/>
    */
-  public setOSMBuildingDisplay(options: { onOff?: boolean }) {
-    if (this.displayStyle.setOSMBuildingDisplay(options))
-      this.synchWithView(false);      // May cahnge frustum depth...
+  public setOSMBuildingDisplay(options: { onOff?: boolean, appearanceOverrides?: FeatureAppearance }) {
+    if (this.displayStyle.setOSMBuildingDisplay(options)) {
+      this._changeFlags.setDisplayStyle();
+      if (options.onOff !== undefined)
+        this.synchWithView(false);      // May change frustum depth...

```

Comment:
Isn't this necessary even if `options.onOff` is `undefined` (meaning toggle current state)?

Reply:
No -- I moved the toggle logic to the keyin -- omitting onOff does not toggle - that seemed too obtuse and didn't allow just changing the appearance overrides.

Reply:
Can you document that?

Reply:
Done - See OsmBuildingDisplayOptions.